csharp/extractor/Semmle.Extraction.CSharp.Standalone/BuildAnalysis.cs

```diff
@@ -45,17 +44,18 @@ interface IBuildAnalysis
     /// </summary>
     class BuildAnalysis : IBuildAnalysis
     {
-        readonly AssemblyCache assemblyCache;
-        readonly NugetPackages nuget;
-        readonly IProgressMonitor progressMonitor;
-        HashSet<string> usedReferences = new HashSet<string>();
-        readonly HashSet<string> usedSources = new HashSet<string>();
-        readonly HashSet<string> missingSources = new HashSet<string>();
-        readonly Dictionary<string, string> unresolvedReferences = new Dictionary<string, string>();
-        readonly DirectoryInfo sourceDir;
-        int failedProjects, succeededProjects;
-        readonly string[] allSources;
-        int conflictedReferences = 0;
+        private readonly AssemblyCache assemblyCache;
+        private readonly NugetPackages nuget;
+        private readonly IProgressMonitor progressMonitor;
+        private HashSet<string> usedReferences = new HashSet<string>();
+        private readonly HashSet<string> usedSources = new HashSet<string>();
+        private readonly HashSet<string> missingSources = new HashSet<string>();
+        private readonly Dictionary<string, string> unresolvedReferences = new Dictionary<string, string>();
+        private readonly DirectoryInfo sourceDir;
+        private int failedProjects, succeededProjects;
+        private readonly string[] allSources;
+        private int conflictedReferences = 0;
+        private readonly object mutex = new object();
```

Comment:
I do actually prefer explicit `private` as it makes the intent clearer.

Unfortunately there's not a `ConcurrentHashSet` so it would mean `ConcurrentDictionary<string,bool>` which is fine but a bit awkward. I'll see how it goes.